app_dart/lib/src/service/gerrit_service.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert' as convert;
+
+import 'package:http/http.dart' as http;
+
+import 'logging.dart';
+
+class GerritService {
+  GerritService({http.Client? httpClient}) : httpClient = httpClient ?? http.Client();
+
+  final http.Client httpClient;
+
+  /// Gets the branches from a remote git repository using the gerrit APIs.
+  Future<List<String>> branches(String repo, String project, String prefix) async {
```

Comment:
Nit: use `match` to avoid confusion with the option `Prefix(p)`.
Nit: add the gerrit api ref link: https://gerrit-review.googlesource.com/Documentation/rest-api-projects.html

Reply:
Updated with subString to match the APIs nomenclature.